src/libsyntax_ext/format.rs

```diff
@@ -43,17 +44,16 @@ struct Context<'a, 'b:'a> {
     /// The span of the format string literal.
     fmtsp: Span,
 
-    /// Parsed argument expressions and the types that we've found so far for
-    /// them.
+    /// List of parsed argument expressions.
+    /// Named expressions are resolved early, and are appended to the end of
+    /// argument expressions.
     args: Vec<P<ast::Expr>>,
-    arg_types: Vec<Option<ArgumentType>>,
-    /// Parsed named expressions and the types that we've found for them so far.
-    /// Note that we keep a side-array of the ordering of the named arguments
-    /// found to be sure that we can translate them in the same order that they
-    /// were declared in.
-    names: HashMap<String, P<ast::Expr>>,
-    name_types: HashMap<String, ArgumentType>,
-    name_ordering: Vec<String>,
+    /// Placeholder slot numbers indexed by argument.
```

Comment:
Could you expand this a bit as well? I found this kinda hard to follow, but it looks like if you have a format string like `{0:x} {0:X} {}` this would be something like:

```
[ 
    [0, 1],
    [0],
]
```

where the indexes are pointers into `arg_unique_types`?